misc/python/materialize/checks/multiple_partitions.py

```diff
@@ -114,7 +120,7 @@ def validate(self) -> Testdrive:
                 > SELECT SUM(p.offset) FROM multiple_partitions_source_progress p;
                 420
 
-                > SELECT status FROM mz_internal.mz_source_statuses WHERE name = 'multiple_partitions_source';
+                >[retry] SELECT status FROM mz_internal.mz_source_statuses WHERE name = 'multiple_partitions_source';
```

Comment:
People are going to get burned going forward that the queries against `mz_internal` will not being retried going forward. Ideally `testdrive` should retry by default any statement containing `mz_internal`.

Such an automatic retry is going to eliminate a lot of flakiness and post-merge fixes going forward.

Reply:
Agreed, but this might also lead to some confusion about why some queries are implicitly retried and others not. So should at least be documented.

Reply:
My plan is to detect this and produce an error saying that retry behavior must be specified so that everything is clear by reading the test